eureka2-write-server/src/main/java/com/netflix/eureka2/server/PeerAddressesProvider.java

```diff
@@ -0,0 +1,71 @@
+package com.netflix.eureka2.server;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Provider;
+import java.util.Arrays;
+
+import com.netflix.eureka2.Server;
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotification.Kind;
+import com.netflix.eureka2.server.config.EurekaServerConfig;
+import com.netflix.eureka2.server.resolver.EurekaEndpoint;
+import com.netflix.eureka2.server.resolver.EurekaEndpoint.ServiceType;
+import com.netflix.eureka2.server.resolver.EurekaEndpointResolvers;
+import com.netflix.eureka2.server.resolver.EurekaEndpointResolvers.ResolverType;
+import rx.Observable;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+
+/**
+ * @author Tomasz Bak
+ */
+public class PeerAddressesProvider implements Provider<Observable<ChangeNotification<Server>>> {
+
+    private final EurekaServerConfig config;
+    private final ServiceType serviceType;
+
+    private Observable<ChangeNotification<Server>> addressStream;
+
+    public PeerAddressesProvider(EurekaServerConfig config, ServiceType serviceType) {
+        this.config = config;
+        this.serviceType = serviceType;
+    }
+
+    public PeerAddressesProvider(Observable<ChangeNotification<Server>> addressStream) {
+        this.config = null;
+        this.serviceType = null;
+        this.addressStream = addressStream;
+    }
+
+    @PostConstruct
+    public void createResolver() {
+        if (config != null) {  // always postConstruct resolve if config exist
```

Comment:
This whole block doesn't seem to do any heavy lifting, why not just move this to the constructor?